x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java

```diff
@@ -41,18 +41,21 @@
             name + "_script_field",
             factoryClass,
             /*
-             * In an ideal world we wouldn't need the script cache at all
-             * because we have a hard reference to the script. The trouble
-             * is that we compile the scripts a few times when performing
-             * a mapping update. This is unfortunate, but we rely on the
-             * cache to speed this up.
+             * We rely on the script cache in two ways:
+             * 1. It caches the "heavy" part of mappings generated at runtime.
+             * 2. Mapping updates tend to try to compile the script twice. Not
+             *    for any good reason. They just do.
```

Comment:
@javanna is in the process of fixing this! Or at least, starting to.

Reply:
Nice.